### ISSUE TITLE
SQL error on site running PostgreSQL when creating diagnostics report

### ISSUE THREAD

**Issue by nicodeboer** [putyourlightson/craft-blitz#724]
### Bug Report

I migrated from MySQL to PostgreSQL and am investigating Blitz issues. But when running 'Generate Diagnostics Report' I'm now getting a 500 server error.

After changing line 161 in vendor/putyourlightson/craft-blitz/src/helpers/DiagnosticsHelper.php to:
```$nestedExpression = new Expression('CASE WHEN "elements_owners"."ownerId" IS NULL THEN 0 ELSE 1 END');```
it all works again as expected.

```Database Exception – [yii\db\Exception](https://www.yiiframework.com/doc-2.0/yii-db-exception.html)

SQLSTATE[42703]: Undefined column: 7 ERROR: column "ownerid" does not exist
LINE 1: ...elementcaches"."elementId") AS "count", CASE WHEN ownerId IS...
^
HINT: Perhaps you meant to reference the column "elements_owners.ownerId".
The SQL being executed was: SELECT "type", count(DISTINCT "elementcaches"."elementId") AS "count", CASE WHEN ownerId IS NULL THEN 0 ELSE 1 END AS "nested"
FROM "blitz_elementcaches" "elementcaches"
INNER JOIN "blitz_caches" ON "elementcaches"."cacheId" = "blitz_caches"."id"
INNER JOIN "elements" ON "elementcaches"."elementId" = "elements"."id"
LEFT JOIN "elements_owners" "elements_owners" ON "elementcaches"."elementId" = "elements_owners"."elementId"
WHERE "siteId"=2
GROUP BY "type", "nested"
ORDER BY "count" DESC, "type"
Error Info: Array
(
    [0] => 42703
    [1] => 7
    [2] => ERROR:  column "ownerid" does not exist
LINE 1: ...elementcaches"."elementId") AS "count", CASE WHEN ownerId IS...
                                                             ^
HINT:  Perhaps you meant to reference the column "elements_owners.ownerId".
)
↵
Caused by: PDOException

SQLSTATE[42703]: Undefined column: 7 ERROR: column "ownerid" does not exist
LINE 1: ...elementcaches"."elementId") AS "count", CASE WHEN ownerId IS...
^
HINT: Perhaps you meant to reference the column "elements_owners.ownerId".```

### Diagnostics Report

#### Application Info

- PHP version: 8.2.24
- Craft edition & version: Pro 5.4.8
- Database driver & version: PostgreSQL 14.13

#### Installed Plugins

- Amazon S3: 2.2.1
- Amazon SES: 3.1.0
- Blitz: 5.9.0
- CKEditor: 4.3.1
- Cookie Consent: 5.0.3
- Craft Commerce: 5.2.0
- Element API: 4.1.0
- Entry Count: 4.0.0
- Feed Me: 6.5.0
- GraphQL Authentication: 3.0.0-RC3
- Knock Knock: 3.0.1
- Mollie for Craft Commerce: 4.2.0
- Olivemenus: 5.0.0
- SendGrid: 3.0.0
- SEOmatic: 5.1.5
- Sprig: 3.5.0
- Super Table: 4.0.2
- Webhooks: 3.1.0

#### Loaded Modules

- yellowbello: Yellowbello\Yellowbello
- mollie-connect: MollieConnect\MollieConnect
- yellowbello-api: YellowbelloApi\YellowbelloApi
- yb-shop-module: ybshopmodule\YbShopModule
- debug: yii\debug\Module
- codeeditor: nystudio107\codeeditor\CodeEditor
- sprig-core: putyourlightson\sprig\Sprig
- verbb-base: verbb\base\Base

#### Blitz Plugin Settings

```json
{
    "debug": true,
    "hintsEnabled": true,
    "cachingEnabled": true,
    "refreshCacheEnabled": true,
    "refreshMode": 3,
    "includedUriPatterns": [
        {
            "enabled": "1",
            "uriPattern": ".*"
        }
    ],
    "excludedUriPatterns": [
        {
            "enabled": "",
            "uriPattern": "\/account\/.*"
        },
        {
            "enabled": "",
            "uriPattern": "\/checkout\/.*"
        },
        {
            "enabled": "",
            "uriPattern": "\/login"
        },
        {
            "enabled": "",
            "uriPattern": "\/register"
        },
        {
            "enabled": "",
            "uriPattern": "\/search"
        },
        {
            "enabled": "",
            "uriPattern": "\/setpassword"
        },
        {
            "enabled": "",
            "uriPattern": "\/signin"
        },
        {
            "enabled": "",
            "uriPattern": "\/cart"
        },
        {
            "enabled": "",
            "uriPattern": "\/robots.txt"
        }
    ],
    "cacheStorageType": "putyourlightson\\blitz\\drivers\\storage\\FileStorage",
    "cacheStorageSettings": {
        "folderPath": "\/var\/www\/vhosts\/yellowbello.nl\/development\/web\/cache\/blitz",
        "compressCachedValues": ""
    },
    "cacheStorageTypes": [],
    "cacheGeneratorType": "putyourlightson\\blitz\\drivers\\generators\\HttpGenerator",
    "cacheGeneratorSettings": {
        "concurrency": "3"
    },
    "cacheGeneratorTypes": [],
    "customSiteUris": [],
    "cachePurgerType": "putyourlightson\\blitz\\drivers\\purgers\\DummyPurger",
    "cachePurgerSettings": [],
    "cachePurgerTypes": [],
    "deployerType": "putyourlightson\\blitz\\drivers\\deployers\\DummyDeployer",
    "deployerSettings": [],
    "deployerTypes": [],
    "ssiEnabled": false,
    "ssiTagFormat": "<!--#include virtual=\"{uri}\" -->",
    "detectSsiEnabled": true,
    "esiEnabled": false,
    "onlyCacheLowercaseUris": false,
    "queryStringCaching": 0,
    "includedQueryStringParams": [
        {
            "enabled": "1",
            "queryStringParam": ".*"
        }
    ],
    "excludedQueryStringParams": [
        {
            "enabled": "1",
            "queryStringParam": "gclid"
        },
        {
            "enabled": "1",
            "queryStringParam": "fbclid"
        }
    ],
    "apiKey": "",
    "generatePagesWithQueryStringParams": true,
    "purgeAssetImagesWhenChanged": true,
    "refreshCacheAutomaticallyForGlobals": false,
    "refreshCacheWhenElementMovedInStructure": true,
    "refreshCacheWhenElementSavedUnchanged": false,
    "refreshCacheWhenElementSavedNotLive": false,
    "cacheNonHtmlResponses": false,
    "trackElements": true,
    "trackElementQueries": true,
    "excludedTrackedElementQueryParams": [],
    "cacheDuration": null,
    "nonCacheableElementTypes": [],
    "sourceIdAttributes": [],
    "liveStatuses": [],
    "integrations": [
        "putyourlightson\\blitz\\drivers\\integrations\\CommerceIntegration",
        "putyourlightson\\blitz\\drivers\\integrations\\SeomaticIntegration"
    ],
    "defaultCacheControlHeader": "no-store",
    "cacheControlHeader": "public, s-maxage=31536000, max-age=0",
    "cacheControlHeaderExpired": "public, s-maxage=5, max-age=0",
    "sendPoweredByHeader": true,
    "outputComments": false,
    "refreshCacheJobPriority": 10,
    "driverJobBatchSize": 100,
    "driverJobPriority": 100,
    "queueJobTtr": 300,
    "maxRetryAttempts": 10,
    "maxUriLength": 255,
    "mutexTimeout": 1,
    "commands": [],
    "injectScriptEvent": "DOMContentLoaded",
    "injectScriptPosition": 3
}
```

#### Recommendations

- ✅ Blitz is configured not to refresh cached pages when an element is saved but unchanged.
- ✅ Blitz is configured not to refresh cached pages when an element is saved but not live.
- ✅ Image transforms are configured to be generated before page load.
- ✅ One or more globals exist and refreshCacheAutomaticallyForGlobals is disabled.
- ✅ The @web alias is explicitly defined.
- ✅ Queue jobs are configured not to run automatically via web requests.
- ✅ The Blitz Hints utility is not reporting any eager-loading opportunities.
- ✅ The Async Queue plugin is not installed or enabled.
- ✅ The blitz/cache/refresh-expired console command has been executed within the past 24 hours.

#### Site Tracking 

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

**Comment by bencroker** [comment]
Thanks for reporting that! Can you please confirm whether this also fixes the issue?

```php
 $nestedExpression = new Expression('CASE WHEN [[ownerId]] IS NULL THEN 0 ELSE 1 END');
```

**Comment by nicodeboer** [comment]
Yes, this also works. Thanks.

Tried it locally also with a MySQL database and there your fix works, but mine does not.

**Comment by bencroker** [comment]
Thanks! Fixed in https://github.com/putyourlightson/craft-blitz/commit/4792c61389bdb028193ecd82c3855e0842830b80 and released in [5.9.1](https://github.com/putyourlightson/craft-blitz/releases/tag/5.9.1).
